formazione/models.py

```diff
@@ -361,6 +365,33 @@ def non_idonei(self):
     def idonei(self):
         return self.partecipazioni_confermate().filter(esito_esame=PartecipazioneCorsoBase.IDONEO)
 
+    def genera_pdf_firme(self):
+        """
+        Genera il fogli firme delle lezioni del corso.
+        """
+
+        iscritti = [x.persona.nome_completo for x in self.partecipazioni_confermate()]
+
+        z = Zip(oggetto=self)
+        for lezione in self.lezioni.all():
+
+            pdf = PDF(oggetto=self)
+            pdf.genera_e_salva(
+                nome="Firme lezione %s.pdf" % lezione.nome,
+                corpo={
+                    "corso": self,
+                    "iscritti": iscritti,
+                    "lezione": lezione,
+                    "data": lezione.inizio.date,
+                },
+                modello="pdf_firme_lezione.html",
+            )
+            z.aggiungi_file(file_path=pdf.file.path, nome_file=pdf.nome)
+
+        z.comprimi_e_salva(nome="Fogli firme %s.zip" % (self.nome, ))
+
+        return redirect(z.download_url)
```

Comment:
In Model-View-Controller (o Model-View-Template, come implementato da Django), il modello deve essere separato dalla vista. In questo caso, `redirect` ritorna una vista, nonostante questo sia il metodo di un modello.

Ti suggerirei di ritornare l'oggetto `Zip`. Se la vista vuole puoi permettere lo scaricamento di tale ZIP, sara' la responsabilita' della stessa vista chiamare `redirect(zip.download_url)`.

Per favore, ove possibile, evita i nomi di variabile con una singola lettera. `zip` e' piu' significativo di `z` (PEP-8).

Reply:
la variabile z è come da esempio che mi hai indicato, userò zip comunque

Reply:
zip sovrascrive builtin zip di python va bene uguale?

Reply:
Suggerirei `archivio`...

Reply:
> la variabile z è come da esempio che mi hai indicato, userò zip comunque

Capisco. Abbi pazienza, non tutto il codice di Gaia e' scritto in modo attinente alle nostre nuove linee guida. Queste ultime sono state scritte due o tre anni fa (mi pare), ma Gaia ha piu' del doppio di anni... :)